Mathlib/Data/Nat/Digits/Lemmas.lean

```diff
@@ -319,4 +319,190 @@ theorem getD_digits (n i : ℕ) {b : ℕ} (h : 2 ≤ b) : (digits b n).getD i 0
+    L ∈ fixedLengthDigits hb l ↔ l = L.length ∧ ∀ x ∈ L, x < b := by
+  simp
+
+@[simp]
+theorem fixedLengthDigits_zero {b : ℕ} (hb : 1 < b) :
+    fixedLengthDigits hb 0 = {[]} := by
+  ext _
+  simpa [eq_comm] using by grind
+
+@[simp]
+theorem fixedLengthDigits_one {b : ℕ} (hb : 1 < b) :
+    fixedLengthDigits hb 1 = Finset.image (fun x : ℕ ↦ [x]) (Finset.range b) := by
+  ext
+  rw [mem_fixedLengthDigits_iff, eq_comm, List.length_eq_one_iff]
+  grind
+
+theorem fixedLengthDigits_card {b : ℕ} (hb : 1 < b) (l : ℕ) :
+    Finset.card (fixedLengthDigits hb l) = b ^ l := by
+  rw [Set.BijOn.finsetCard_eq (ofDigits b) (bijOn_ofDigits' hb l), Finset.card_range]
+
+/--
+Let `L` be a list in `List.fixedLengthDigits b l`. Then `L` is equal to the list of digits of
+`ofDigits b L` concatenated with enough `0`'s to that its length is equal to `l`.
+-/
+theorem eq_digits_ofDigits_append_of_mem_fixedLengthDigits {b : ℕ} (hb : 1 < b) (l : ℕ)
+    {L : List ℕ} (hL : L ∈ fixedLengthDigits hb l) :
+    L = b.digits (ofDigits b L) ++ replicate (l - (b.digits (ofDigits b L)).length) 0 := by
+  refine (Nat.bijOn_ofDigits' hb l).2.1 hL ?_ (by simp [Nat.ofDigits_digits])
+  refine (mem_fixedLengthDigits_iff hb).mpr ⟨?_, fun x hx ↦ ?_⟩
+  · rw [length_append, length_replicate, Nat.add_sub_cancel']
+    rw [digits_length_le_iff hb, ← Finset.mem_range]
+    exact (Nat.bijOn_ofDigits' hb l).1 hL
+  · simp only [mem_append, mem_replicate] at hx
+    obtain hi | ⟨_, rfl⟩  := hx
+    · exact digits_lt_base hb hi
+    · linarith
+
+/--
+The `Finset` of lists whose head is a fixed integer `d` and tail is a list
+in `List.fixedLengthDigits b l`.
+-/
+noncomputable abbrev consFixedLengthDigits {b : ℕ} (hb : 1 < b) (l d : ℕ) :
+    Finset (List ℕ) := Finset.image (fun L ↦ d::L) (fixedLengthDigits hb l)
+
+
+theorem ne_empty_of_mem_consFixedLengthDigits {b : ℕ} (hb : 1 < b) {l d : ℕ} {L : List ℕ}
+    (hL : L ∈ consFixedLengthDigits hb l d) : L ≠ [] := by
+  obtain ⟨_, _, rfl⟩ := Finset.mem_image.mp hL
+  exact cons_ne_nil d _
+
+theorem consFixedLengthDigits_head {b : ℕ} (hb : 1 < b) {l d : ℕ} {L : List ℕ}
+    (hL : L ∈ consFixedLengthDigits hb l d) :
+    List.head L (ne_empty_of_mem_consFixedLengthDigits hb hL) = d := by
+  obtain ⟨_, _, rfl⟩ := Finset.mem_image.mp hL
+  rw [head_cons]
+
+/--
+If `L` is a list in `List.fixedLengthDigits b l` and `d` is an integer `< b`, then
+`d::L` is a list in `List.fixedLengthDigits b (l + 1).`
+-/
+theorem cons_mem_fixedLengthDigits_succ {b : ℕ} (hb : 1 < b) (l d : ℕ) (hd : d < b) {L : List ℕ}
+    (hL : L ∈ fixedLengthDigits hb l) :
+    d::L ∈ fixedLengthDigits hb (l + 1) := by
+  refine (mem_fixedLengthDigits_iff hb).mpr ⟨?_, ?_⟩
+  · simpa using ((mem_fixedLengthDigits_iff hb).mp hL).1
+  · intro x hx
+    obtain rfl | hx := mem_cons.mp hx
+    · exact hd
+    · exact ((mem_fixedLengthDigits_iff hb).mp hL).2 _ hx
+
+theorem disj_consFixedLengthDigits {b : ℕ} (hb : 1 < b) (l : ℕ) :
```

Comment:
I don't think there is an established convention to use `disj` for `PairwiseDisjoint`, so I would suggest just going with the name according to the naming convention:

```suggestion
theorem pairwiseDisjoint_consFixedLengthDigits {b : ℕ} (hb : 1 < b) (l : ℕ) :
```